pkg/platform/kube/functionres/lazy.go

```diff
@@ -1532,7 +1532,10 @@ func (lc *lazyClient) generateCronTriggerCronJobSpec(functionLabels labels.Set,
 	functionAddress := fmt.Sprintf("%s:%d", host, port)
 
 	// generate the curl command to be run by the CronJob to invoke the function
-	curlCommand := fmt.Sprintf("curl --silent %s %s", headersAsCurlArg, functionAddress)
+	// invoke the function (retry for 10 seconds)
+	curlCommand := fmt.Sprintf("curl --silent %s %s --retry 10 --retry-delay 1 --retry-max-time 10 --retry-connrefused",
```

Comment:
will it try on `connrefused` only? or any err between 400 and 500?

Reply:
--retry-connrefused

In addition to the other conditions, consider ECONNREFUSED as a transient error too for --retry. This option is used together with --retry.